packages/gatsby/src/services/recompile.ts

```diff
@@ -24,4 +24,26 @@ export async function recompile({
     // Suspending is just a flag, so it's safe to re-suspend right away
     webpackWatching.suspend()
   })
+
+  const ssrBundlePromise = serverDataDirty
+    ? updateSSRBundle(context)
+    : Promise.resolve()
+
+  const [stats] = await Promise.all([devBundlePromise, ssrBundlePromise])
+  return stats
+}
+
+async function updateSSRBundle({
+  program,
+  websocketManager,
+}: IBuildContext): Promise<void> {
+  reporter.verbose(`Recompiling SSR bundle`)
+
+  const { close } = await buildRenderer(program, Stage.DevelopHTML)
```

Comment:
I wonder if we should also return compiler from `doBuildRenderer` so we can do a simple run and not have to close the compiler each time. This should make rebuilds a little bit faster. (This can be follow up)

Reply:
My webpack-foo is not perfect, so any improvements are welcome in a follow-up ;)

Reply:
But also [this page](https://webpack.js.org/api/node/#compiler-instance) has a quote which makes me think that you can't start multiple compilations?

> The API only supports a single concurrent compilation at a time. When using run or watch, call close and wait for it to finish before calling run or watch again. Concurrent compilations will corrupt the output files.

Reply:
oh okay, I see. Seems like this got changed with FS caching. Thanks for looking it up!